src/clients/storage/StorageClientBase.h

```diff
@@ -114,7 +114,7 @@ class StorageRpcResponse final {
 /**
  * A base class for all storage clients
  */
-template <typename ClientType>
+template <typename ClientType, typename ClientManagerType>
```

Comment:
I need different implemention of `ClientManger` since `ThriftClientManager` is only for different `ThriftClientType`, and there are some implementions binded to client connection. So I need another Manager to do local function call. 
If you have a better idea,  I'm glad to change this. XD